lib/findRelationsInSourceCode.js

```diff
@@ -1,66 +1,42 @@
-// const esprima = require("esprima");
-const parser = require("@babel/parser");
-const traverse = require("@babel/traverse").default;
+const importRegExp = /import.+(["'])((?:\\\1|.)*?)\1/;
+const requireRegExp = /require\((["'])((?:\\\1|.)*?)\1\)/;
 
-function isRequireCallExpression(node) {
-  return (
-    node.type === "CallExpression" &&
-    node.callee.type === "Identifier" &&
-    node.callee.name === "require"
-  );
-}
+function findAllRequires(source) {
+  const requires = [];
+
+  let remainingString = source;
+  let match = remainingString.match(requireRegExp);
+
+  while (match) {
+    requires.push(match[2]);
+    remainingString = remainingString.substr(match.index + match[0].length);
+    match = remainingString.match(requireRegExp);
+  }
 
-function isImportDeclaration(node) {
-  return node.type === "ImportDeclaration";
+  return requires;
 }
 
-module.exports = function findRelationsInSourceCode(source) {
-  const relations = [];
+function findAllImports(source) {
+  const requires = [];
 
-  const ast = parser.parse(source, {
-    // https://babeljs.io/docs/en/next/babel-parser.html#options
-    // Will make @babel/parser guess based on the presence of import/export
-    sourceType: "unambiguous",
-    // https://babeljs.io/docs/en/next/babel-parser.html#plugins
-    plugins: [
-      // Enable objectRestSpread plugin.
-      "objectRestSpread",
-      // Enable the JSX syntax plugin
-      "jsx"
-    ]
-  });
+  let remainingString = source;
+  let match = remainingString.match(importRegExp);
 
-  traverse(ast, {
-    enter(path) {
-      const { node } = path;
+  while (match) {
+    requires.push(match[2]);
+    remainingString = remainingString.substr(match.index + match[0].length);
+    match = remainingString.match(importRegExp);
+  }
 
-      if (isRequireCallExpression(node)) {
-        if (node.arguments.length !== 1) {
-          // line number = meta.start.line
-          throw new Error(
-            "Cannot parse require calls with more than one argument." +
-              "\n\n    " +
-              source.slice(node.loc.start.offset, node.loc.end.offset)
-          );
-        }
-        if (node.arguments[0].type !== "StringLiteral") {
-          throw new Error(
-            "Can only parse require calls with literal values." +
-              "\n\n    " +
-              source.slice(node.loc.start.offset, node.loc.end.offset)
-          );
-        }
-        relations.push(node.arguments[0].value);
-      }
+  return requires;
+}
 
-      if (isImportDeclaration(node)) {
-        // If an import declaration does not have a StringLiteral source it will
-        // throw a Syntax Error "Unexpected token."
+module.exports = function findRelationsInSourceCode(source) {
+  const imports = findAllImports(source);
 
-        relations.push(node.source.value);
-      }
-    }
-  });
+  if (imports.length > 0) {
+    return imports;
```

Comment:
This is the only bit I'm unusure about - I think you can still use require on node even in modules.

Reply:
You're not supposed to be allowed to do that with native ESM in node core. With the esm module you can though - and webpack probably also allows you to.

If it ever comes up as an actual problem we can refactor the method to return `[...requires, ...imports]`, but since most people are either doing one or the other we can assume that we don't need to check for cjs when we found esm.

Reply:
Hmm I believe mutual exlusivity was relaxed - one direction is allowed the other is not (last I read). Unfortuntely, I also think I've written code like that :/